commit caecf2a0ce760f03a7f5ffd63c9001cc1cfc2fbf
Author: Andreas Schwab <schwab@suse.de>
Date:   Tue Nov 26 12:22:58 2013 +0100

    Fix CFI annotations in pthread_cond_timedwait for i486+

2013-12-09  Andreas Schwab  <schwab@suse.de>

	[BZ #15843]
	* sysdeps/unix/sysv/linux/i386/i486/pthread_cond_timedwait.S
	(__pthread_cond_timedwait): Remove wrong cfi_adjust_cfa_offset
	before __condvar_tw_cleanup2 label.

diff --git a/nptl/sysdeps/unix/sysv/linux/i386/i486/pthread_cond_timedwait.S b/nptl/sysdeps/unix/sysv/linux/i386/i486/pthread_cond_timedwait.S
index a6d6bc4..b74e4b5 100644
--- a/nptl/sysdeps/unix/sysv/linux/i386/i486/pthread_cond_timedwait.S
+++ b/nptl/sysdeps/unix/sysv/linux/i386/i486/pthread_cond_timedwait.S
@@ -505,9 +505,9 @@ __pthread_cond_timedwait:
 #endif
 	call	__lll_unlock_wake
 	jmp	11b
-	cfi_adjust_cfa_offset(-FRAME_SIZE)
 
 #ifndef __ASSUME_FUTEX_CLOCK_REALTIME
+	cfi_adjust_cfa_offset(-FRAME_SIZE)
 .Lreltmo:
 	/* Get internal lock.  */
 	movl	$1, %edx
@@ -742,7 +742,6 @@ __pthread_cond_timedwait:
 # endif
 	call	__lll_lock_wait
 	jmp	106b
-	cfi_adjust_cfa_offset(-FRAME_SIZE)
 #endif
 
 	.size	__pthread_cond_timedwait, .-__pthread_cond_timedwait
